criu/kerndat.c

```diff
@@ -1153,29 +1155,59 @@ int kerndat_has_nspid(void)
 static int kerndat_has_nftables_concat(void)
 {
 #if defined(CONFIG_HAS_NFTABLES_LIB_API_0) || defined(CONFIG_HAS_NFTABLES_LIB_API_1)
+	pid_t pid;
 	struct nft_ctx *nft;
-	int ret = 0;
+	int ret, status;
 
-	nft = nft_ctx_new(NFT_CTX_DEFAULT);
-	if (!nft)
+	pid = fork();
+	if (pid < 0) {
+		pr_perror("Can't fork");
 		return -1;
+	}
 
-	if (NFT_RUN_CMD(nft, "create table inet CRIU-kerndat-test")) {
-		ret = -1;
-		goto nft_ctx_free_out;
+	if (pid == 0) {
+		/*
+		 * Create a separate network namespace to avoid
+		 * collisions between two CRIU instances.
+		 */
+		if (unshare(CLONE_NEWNET)) {
+			pr_perror("Unable create a network namespace");
+			exit(-1);
+		}
+
+		nft = nft_ctx_new(NFT_CTX_DEFAULT);
+		if (!nft)
+			exit(-1);
+
+		if (NFT_RUN_CMD(nft, "create table inet CRIU")) {
+			ret = -1;
+			pr_err("Can't create nftables table\n");
+			goto nft_ctx_free_out;
+		}
+
+		if (NFT_RUN_CMD(nft, "add set inet CRIU conn { type ipv4_addr . inet_service ;}"))
+			ret = false; /* kdat.has_nftables_concat = false */
+		else
+			ret = true; /* kdat.has_nftables_concat = true */
+
+		/* Clean up */
+		NFT_RUN_CMD(nft, "delete table inet CRIU");
+
+	nft_ctx_free_out:
+		nft_ctx_free(nft);
+		exit(ret);
 	}
 
-	if (NFT_RUN_CMD(nft, "add set inet CRIU-kerndat-test conn { type ipv4_addr . inet_service ;}"))
-		kdat.has_nftables_concat = false;
-	else
-		kdat.has_nftables_concat = true;
+	if (waitpid(pid, &status, 0) != pid) {
+		pr_perror("Unable to wait %d", pid);
+		return -1;
+	}
 
-	/* Clean up */
-	NFT_RUN_CMD(nft, "delete table inet CRIU-kerndat-test");
+	if (status < 0)
+		return -1;
```

Comment:
This condition for `status` doesn't work as expected. For example, with the following change:
```diff
--- a/criu/kerndat.c
+++ b/criu/kerndat.c
@@ -1195,6 +1195,7 @@ static int kerndat_has_nftables_concat(void)
 
        nft_ctx_free_out:
                nft_ctx_free(nft);
+               ret = -1;
                exit(ret);
        }
```

it shows:

```bash
$ sudo rm -f /run/criu.kdat && sudo criu check --feature network_lock_nftables
network_lock_nftables is supported
```